web/src/components/Post/PostForm/PostForm.tsx

```diff
@@ -297,25 +299,54 @@ const PostForm = (props: PostFormProps) => {
                     })
                   }
                 />
+                <MediaLibrary
+                  name="coverImage"
+                  handleMediaLibrary={([{ public_id, secure_url }]) =>
+                    setCoverImage({
+                      imageId: public_id,
+                      url: secure_url,
+                    })
+                  }
+                />
+                </div>
+                <span className="rw-label"> Image Gallery </span>
+                <div className='flex flex-row flex-wrap gap-2'>
                 <Upload
                   name="imageGalleries"
                   multiple={true}
                   handleUpload={({ ...images }) => {
                     handleSetImageGalleries(images)
                   }}
                 />
+                <MediaLibrary
+                  name="imageGalleries"
+                  handleMediaLibrary={({ ...images }) => {
+                    handleSetImageGalleries(images as ICloudinaryUploadResultInfo[])
+                  }}
+                />
               </div>
             </>
           )}
 
           {postType === EPostType.PHOTO_GALLERY && (
-            <Upload
-              name="imageGalleries"
-              multiple={true}
-              handleUpload={({ ...images }) => {
-                handleSetImageGalleries(images as ICloudinaryUploadResultInfo[])
-              }}
-            />
+            <>
+              <span className="rw-label"> Image Gallery </span>
+              <div className='flex flex-row flex-wrap gap-2'>
+                <Upload
+                  name="imageGalleries"
+                  multiple={true}
+                  handleUpload={({ ...images }) => {
+                    handleSetImageGalleries(images as ICloudinaryUploadResultInfo[])
+                  }}
+                />
+                <MediaLibrary
+                name="imageGalleries"
+                handleMediaLibrary={({ ...images }) => {
+                  handleSetImageGalleries(images as ICloudinaryUploadResultInfo[])
```

Comment:
Shouldn't this then be of the type `ICloudinaryMediaLibraryAsset`, or are they the same? And if they are the same, shouldn't we then only use one type?